frontend/src/metabase/lib/entities.js

```diff
@@ -309,6 +309,7 @@ export function createEntity(def) {
       withCachedDataAndRequestState(
         entityQuery => [...getListStatePath(entityQuery)],
         entityQuery => [...getListStatePath(entityQuery), "fetch"],
+        entityQuery => entity.getQueryKey(entityQuery),
```

Comment:
> Fix loader flicker

Actually for me it's more than just a flicker, I cannot browse the first database - let's add a repro for it?

With 10 DBs connected:
1. Open http://localhost:3000/admin/datamodel/segments
2. Click "Filter by table"
3. Click first DB
4. _Allow things to reload_
5. _First DB now has chevron down icon, "Raw Data" option appeared_
6. Click first DB

It does not open. Things reload but state does change. You have to click 2nd DB to open the 1st one.

Recorded in `master`: 

https://github.com/user-attachments/assets/0becf738-128d-4294-8db2-4b37a0f5c503

Reply:
Great catch, added a e2e test